lib/urlwatch/jobs.py

```diff
@@ -154,6 +155,26 @@ def get_guid(self):
         sha_hash.update(location.encode('utf-8'))
         return sha_hash.hexdigest()
 
+    def request_resources(self, resources):
+        """Request external resources.
+
+        Check if required resources is available in `resources` (a dict).
+        If not, request resources and save them in `resources`. Keys for
+        resources should typically be `self` or class, depending on the
+        shared or exclusive nature of the resource.
+
+        This method should be called sequentially on the main thread to
+        ensure success.
+
+        Args:
+            resources: A dict storing external resources.
+        """
+        ...
+
+    def release_resources(self, resources):
+        """Release external resources requested in `request_resources`."""
+        ...
```

Comment:
Sounds like maybe a context manager is the right thing to do here?

Reply:
The idea crossed my mind. Can't remember why I decided against it. I'll look into it.